hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/DeleteHoodieTableCommand.scala

```diff
@@ -36,9 +37,13 @@ case class DeleteHoodieTableCommand(deleteTable: DeleteFromTable) extends Hoodie
 
     // Remove meta fields from the data frame
     var df = removeMetaFields(Dataset.ofRows(sparkSession, table))
-    if (deleteTable.condition.isDefined) {
-      df = df.filter(Column(deleteTable.condition.get))
+    // SPARK-38626 DeleteFromTable.condition is changed from Option[Expression] to Expression in Spark 3.3
+    val condition: Expression = deleteTable.condition match {
```

Comment:
Instead of doing that let's modify this class to hold the `Option[Expression]` instead of `DeleteFromCommand`

Reply:
If modified like that, I think we would still have to handle the type matching at L133 HoodieAnalysis.scala

Reply:
The overarching idea is to limit this compatibility code to live inside Spark-specific domains (SparkAdapter, HoodieCatalystExpressionUtils, etc). 

> If modified like that, I think we would still have to handle the type matching at L133 HoodieAnalysis.scala

Good call. We can modify then previous suggestion and instead of `resolveDeleteFromCommand` we can either do `extractCondition(DeleteFromCommand)`

Reply:
Several things I've tried:
- exctract `Expression` from `condition`
- Embed `condition` to `Option[Expression]`
Both approches above would make `DeleteHoodieTableCommand` cleaner but have the same problem: we have to provide condition back to `DeleteFromTable` around L417 in `HoodieAnalysis`, which means we have to wrap/unwrap it once more for different spark versions. 

If we want to also make `DeleteHoodieTableCommand` cleaner, I think the only way might be adding two different methods in `SparkAdapter`: one to `resolveDeleteFromTable`, another to `extractCondition`

Reply:
Several things I've tried:
- exctract `Expression` from `condition`
- Embed `condition` to `Option[Expression]`
Both approches above would make `DeleteHoodieTableCommand` cleaner but have the same problem: we have to provide condition back to `DeleteFromTable` around L417 in `HoodieAnalysis`, which means we have to call adapter once more to wrap/unwrap for different spark versions. 

If we also want to also make `DeleteHoodieTableCommand` cleaner, I think the only way might be adding two different methods in `SparkAdapter`: one to `resolveDeleteFromTable`, another to `extractCondition`